app/scripts/modules/titus/src/instance/details/instance.details.controller.js

```diff
@@ -309,9 +309,33 @@ module(TITUS_INSTANCE_DETAILS_INSTANCE_DETAILS_CONTROLLER, [
       return Object.keys($scope.instance.resources.ports).length > 0;
     };
 
+    const constructTaskActions = (instance) => {
+      const constantActions = [
+        { label: 'Terminate', triggerAction: this.terminateInstance },
+        { label: 'Terminate and Shrink Server Gorup', triggerAction: this.terminateInstanceAndShrinkServerGroup },
+      ];
+      const conditionalActions = [];
+
+      if (this.canRegisterWithDiscovery()) {
+        conditionalActions.push({
+          label: 'Enable In Discovery',
+          triggerAction: this.enableInstanceInDiscovery,
+        });
+      }
```

Comment:
I noticed that `instance` is not used in the constructActions block and also that the actions themselves are referenced off the `this` object.  Do you have a plan to refactor this such that the actions can be decoupled from from the instanceDetailsController and managed separately?  i.e., something like:

```
import { enableInstanceInDiscoveryAction }  from 'somewhere'
     if (this.canRegisterWithDiscovery()) {
        conditionalActions.push(instance => enableInstanceInDiscoveryAction(instance));
     }
```

Reply:
Yes, that is exactly what I was thinking. Eventually I will make this a functional React component, where there is no `this` object and the component only captures the conditional logic to construct the actions list. The action functions will be kept either in core or cloudProvider specific folders.

I will remove the `instance` from here so its not confusing for now.